### PR TITLE
Sola Implement Summary Management Features

### PR DESCRIPTION
**Pull request by OAkinbode**
# Description
This PR completes the work started on PR 838 (Summary Management). It implements all the features required in the front end to allow for a summary receiver (manager or mentor) to review the weekly summaries of everyone on their summary team in a simple and quick way. It also implements a way to see which summary teams a volunteer belongs to under "profile". Finally, it implements a way for an admin to create, modify or delete summary teams and populate the fields with volunteers and managers/mentors.

## Related PRS (if any):
This frontend PR is related to the #358 backend PR and continues the work on #838 on the front end.
…

## Main changes explained:
The PR implements CRUD operations for Summary Management functions.
…

## How to test:
1. check into the current branch on the front end and PR 358 on the backend
2. do pm install and nom run start:local to run this PR locally
3. watch the attached demo video to see what features are included and should be tested.
4. log in as Admin and check that "Summary Management" shows up under "other links" in the nav bar. Also check that you can add, remove and modify summary teams. Create teams and populate the team members and summary receiver fields.
5. Go under "Profile" in any role and see that the user can see the summary teams to which they belong under "Summary
Teams" tab.
6. Create an account with Manager or Mentor role and check that you can see Summary Management in the nav bar. Check that you can view reports of all teams in which the manager and mentor user is a summary receiver.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/a6ccfb79-114b-4693-9190-a21d15196a87

## Note:
Though, the PR has more changes than usual to test. Some of the tests listed in the video have been tested in previous PRs e.g., PR #838.

### CODE REVIEW

**Review by joyceiphone — COMMENTED** [review]
@OAkinbode. How to get to see summary management? I logged in as an administrator and couldn't see summary management from the dropdown menu.

**Review by joyceiphone — COMMENTED** [review]
Screen Shot 2023-07-22 at 9 25 10 PM
Unable to add new summary team.

**Review by winghojackyli — DISMISSED** [review]
I tested the PR and it works great on my side. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/532adfba-9948-4ddd-85d1-095053a43d67

**Comment by OAkinbode** [comment]
> Screen Shot 2023-07-22 at 9 25 10 PM Unable to add new summary team.

Hello @joyceiphone 
Screenshot 2023-07-22 at 7 55 05 PM
Thank you for the comment. I just ran the application locally and checked the summary page. It appears someone, maybe you, successfully tested that feature (I did not create JackyTestingSummaryGroup). I also tested it again myself and it works fine. Please check that you are properly checked in on the front end to PR 1033 and on the back end to PR 358. If it still fails on your end, please record a video or give some more detail. Thanks.

**Review by n0vay — CHANGES_REQUESTED** [review]
I tested the website and it works, Great Job on the code!
The summary management only shows up for Team Manager and not team Mentor or any other role even if the account is added as reciever for the summary.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/85691749/3b100555-97c7-4c4e-83be-c4f4289d4cbe

**Review by shiva2096 — COMMENTED** [review]
Hi @OAkinbode 

I have checkout this PR locally, and the backend PR 358.
As soon as I open the dashboard, I have getting following error in the backend console.

Screenshot 2023-07-23 at 12 09 03 AM

Because of this I am not able test the summary management functionality, not able to add members to the Summary Group.

Please let me know if you need any other details regrading the issue.

**Comment by n0vay** [comment]
I am also getting the following errors in console for backend.
image

**Comment by OAkinbode** [comment]
> @OAkinbode

Hello @n0vay, Thank you for catching that. I have pushed an update that addresses this issue.

**Comment by sxiong5** [comment]
Hey @OAkinbode, I have the same issue here with @joyceiphone. I am pretty sure I have switched to the [HGNRest#358](https://github.com/OneCommunityGlobal/HGNRest/pull/358) and keep the code up to date, but I still receive 404.

Screenshot 2023-07-24 at 15 08 50

Screenshot 2023-07-24 at 15 09 07

Please point out if I did something wrong.

**Comment by OAkinbode** [comment]
> Hey @OAkinbode, I have the same issue here with @joyceiphone. I am pretty sure I have switched to the [HGNRest#358](https://github.com/OneCommunityGlobal/HGNRest/pull/358) and keep the code up to date, but I still receive 404.
> 
> Screenshot 2023-07-24 at 15 08 50
> Screenshot 2023-07-24 at 15 09 07
> 
> Please point out if I did something wrong.

Hello @sxiong5 and @joyceiphone . Thank you for the feedback. I have pushed an update that fixes this problem. Please pull the latest changes on the frontend (#1033). You should not have this problem again. If you do, please let me know.

**Review by sxiong5 — CHANGES_REQUESTED** [review]
Hey @OAkinbode, I have retested this PR. The 404 error has been solved! But there is a new issue. The list of groups didn't show up. In the video, I created a group and got 200 from the backend, which means creating successfully! But after I refreshed the page, it didn't display.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/603c97b1-76dc-4edb-a4bb-2a8ade54055d

**Comment by OAkinbode** [comment]
> Hey @OAkinbode, I have retested this PR. The 404 error has been solved! But there is a new issue. The list of groups didn't show up. In the video, I created a group and got 200 from the backend, which means creating successfully! But after I refreshed the page, it didn't display.
> 
>  Screen.Recording.2023-07-28.at.14.44.46.mov

Hello @sxiong5. Thank you for your feedback. I am trying to replicate the error. Running the latest from the back and the front end, this is how it works at my end: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/758bf146-4213-4887-99cb-9d879f32aff2

Is it possible to give me more information so I can trace the issue? I will also look out to see if others find the same issue.

**Review by shiva2096 — COMMENTED** [review]
Hi @OAkinbode 

I have restest the PR and I still see the same error in backend console, when I open the dashboard of any user.

Screenshot 2023-07-29 at 12 28 26 PM

Here is what I did:
1) checked out to 358 backend branch
2) checked out to 1033 frontend branch
3) Started everything as usual.

Please let me know if you need any other details.

**Review by zuhang2000 — CHANGES_REQUESTED** [review]
Hi I tested your PR by following your instructions. I checked out both #358 and #1033 branch. And I did the test in three different roles which are Admin, Manager, Volunteer. 

For Admin account, I tried to create two new summary groups and add Volunteer user and Admin user to the groups. I also add Manager user to the groups. The operation went successfully and the groups are shown in the list. One question is that I am not sure if it's the case that user should not be able to see the summary teams they are in if the user is a summary receiver in that team. Because I notice that in the profile page, Manager user can not see any summary teams they are in.

I also test creating duplicate summary team name, adding the same user several times, and deleting users from the team. These operations are successful and work fine. I also test that the manager is able to see the summary reports if they are a summary receive in the corresponding team. Manager is also able to delete a team member and then the summary report of that deleted team member is no longer in the list. 

All these functionality works great.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/123051008/8c3f6399-7eef-453a-bad3-84369d465943

 

But I do notice that whenever I got redirected to the dashboard page, the following error is reported in the backend. I test for different roles and they all got the same problem each time. Please take a look at it. 
Screen Shot 2023-07-29 at 4 07 20 PM

**Review by iTvX — DISMISSED** [review]
In the first video, I create a manager user and assign the team. After I log in to that manager user account, everything is as you described.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/f6401608-133e-40bb-8b8c-31112d10f0b7

For the second video, I continued using the new manager account. However, when I clicked on "team management," no teams were showing up. I attempted to clean the cookies, but it did not resolve the issue. Although I could add a team, the team name was not displayed at all. Interestingly, when I switched back to my admin account, everything worked fine. Therefore, I believe this is a bug that needs to be fixed.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46791134/4a2c4210-c710-4934-849c-ee4e746b1b68

**Comment by OAkinbode** [comment]
> 

Thank you @iTvX for your comment. I checked the account you mentioned. I saw that both your accounts were under "summary team" and none was under the "summary receiver". A manager will only see teams of which they are summary receivers. Since there is no account set as a summary receiver, that team will not show up on your summary dashboard. See the video below. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/c050073e-14bc-45c8-8b8d-4e3180213261

Thanks again.

**Comment by iTvX** [comment]
> > 
> 
> Thank you @iTvX for your comment. I checked the account you mentioned. I saw that both your accounts were under "team member" and none was under the "summary receiver". A manager will only see teams of which they are summary receivers. Since there is no account set as a summary receiver, that team will not show up on your summary dashboard. See the video below.
> 
>  Screen.Recording.2023-08-04.at.6.13.56.PM.mov 
> Thanks again.

Thank you for the reply, my mistake. I have tried your way, it works.

**Review by ThapeloMasasa — COMMENTED** [review]
The creation of the summary Group is not working and I receive the errors below in the console.
Screenshot 2023-08-07 at 4 34 11 PM

**Review by iTvX — DISMISSED** [review]
Tested again, everything is good now

**Review by winghojackyli — DISMISSED** [review]
Hi @OAkinbode , I've reviewed the updated PR and it works as intended. 
I can do CRUD actions on the summary group and view report. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108389005/ac56c3f5-d485-45cb-af5d-ceb80e574a29

**Comment by veronicacheng2** [comment]
> The creation of the summary Group is not working and I receive the errors below in the console. Screenshot 2023-08-07 at 4 34 11 PM

Hi @OAkinbode, I have a similar situation as @ThapeloMasasa
image

File: src/components/SummaryManagement/summaryGroupTableBody.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    return false;
+  };
+
+  const updateinfo = async roles => {
+    const id = props.summaryGroupId;
+    if (id && !roles) {
+      await props.updateUserInfo(id);
+    }
+  };
+  if (isReceiver || props.currentUserRole === 'Administrator') {
```
Comment (sxiong5) [inline src/components/SummaryManagement/summaryGroupTableBody.jsx:43]: > Hey @OAkinbode, I have retested this PR. The 404 error has been solved! But there is a new issue. The list of groups didn't show up. In the video, I created a group and got 200 from the backend, which means creating successfully! But after I refreshed the page, it didn't display.
> 
>  Screen.Recording.2023-07-28.at.14.44.46.mov

Hey @OAkinbode, I found what my issue is. I tested the PR with Owner account. But this line of code only makes it available to the admin. That's why I cannot see the table content. But this also confused me as to why an owner can not see the list but is allowed to create it. Please correct me if I am wrong!

Also, a reminder can be added here if the user doesn't have access to the table.

**Comment by beblicarl** [comment]
Hello, I tested this functionality and everything works as intended. the code is great as well

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.08.16-11_44_21.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/ddad567e-1eeb-4220-9ba5-9d44c59cd6c0)

**Review by beblicarl — DISMISSED** [review]

File: src/components/SummaryManagement/summaryGroupTableBody.jsx (hunk 1 above)
Comment (OAkinbode) [reply src/components/SummaryManagement/summaryGroupTableBody.jsx:43]: Hello @sxiong5. Thank you for the review. The way it works at the moment, only an admin can fully create/approve a group at the initial stage. A manager/mentor might be able to create the group, but they can have access only after an admin adds them as the summary reviewer. After that, they could add other members as needed. The intention is to control the creation of groups, but allow for managers/mentors to have some input. That workflow may be changed at some point in the future as Jae and other managers see fit.

**Review by DavidC0126 — APPROVED** [review]
Hi @OAkinbode I tested this PR with backend PR#358 and they all work as intended.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/56d8809f-dc02-48e2-8125-79b98a0b242e

**Comment by Lin1404** [comment]
Hi @OAkinbode , I tested this PR, and I use development backend since PR#358 is merged. And I got 404 error.

image

And I have checked the backend code. I think the reason for this error is API missing in backend.
image

File: src/components/SummaryManagement/summaryGroupTableBody.jsx (hunk 1 above)
Comment (sxiong5) [reply src/components/SummaryManagement/summaryGroupTableBody.jsx:43]: Yes, I agree. But should the Owner and Admin have the same permission? My issue is that I cannot access the group list as an Owner (which should have the highest permission).

**Comment by liaoyufu-leo** [comment]
Hi @OAkinbode , I tested this PR, and I use development backend since PR#358 is merged. And I got 404 error after I want to create a summary group.
image

**Comment by OAkinbode** [comment]
> Hi @OAkinbode , I tested this PR, and I use development backend since PR#358 is merged. And I got 404 error after I want to create a summary group. image

Hello @liaoyufu-leo and @Lin1404 . Thank you for reviewing the PR. I am not sure PR 1033 will work with the development Backend because PR 358 on HGNRest is not merged yet with development. See my page showing the state of backend PR 358 at the moment. I think it should work if you pull this backend PR. If I am missing something else, please let me know. 
Screenshot 2023-08-18 at 9 32 31 PM

File: src/components/SummaryManagement/summaryGroupTableBody.jsx (hunk 1 above)
Comment (one-community) [reply src/components/SummaryManagement/summaryGroupTableBody.jsx:43]: Yes, I agree. Owner should have same permission as Admin. Please update to include that.

**Review by sxiong5 — DISMISSED** [review]
Hey @OAkinbode, I have tested the PR and everything LGTM now.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/62367016/8e39187e-c966-4097-9064-e678cccd286b

**Comment by Shubhu0** [comment]
Checked the PR, works as expected.
